vars/dockerExecuteOnKubernetes.groovy

```diff
@@ -244,13 +254,24 @@ private String stashWorkspace(config, prefix, boolean chown = false) {
             sh """#!${config.containerShell?:'/bin/sh'}
 chown -R ${runAsUser}:${fsGroup} ."""
         }
-        stash(
-            name: stashName,
-            includes: config.stashIncludes.workspace,
-            excludes: config.stashExcludes.workspace,
-            //inactive due to negative side-effects, we may require a dedicated git stash to be used
-            //useDefaultExcludes: false
-        )
+
+        if (stashBack) {
+            stash(
+                name: stashName,
+                includes: config.stashIncludes.stashBack ?: config.stashIncludes.workspace,
+                excludes: config.stashExcludes.stashBack ?: config.stashExcludes.workspace,
+                //inactive due to negative side-effects, we may require a dedicated git stash to be used
+                //useDefaultExcludes: false
+            )
+        } else {
+            stash(
+                name: stashName,
+                includes: config.stashIncludes.workspace,
+                excludes: config.stashExcludes.workspace,
+                //inactive due to negative side-effects, we may require a dedicated git stash to be used
+                //useDefaultExcludes: false
+            )
+        }
```

Comment:
Wouldn't it be easier to read if we keep the old stash  (without boolean `stashBack` and the corresponding `if/else`) and change the includes and excludes to the elvis version? In case the stashBack includes and excludes are defined they are used, otherwise we keep the old config parameters.

Reply:
not really sure what you mean ...

We need a different behavior into the container vs. out of the container.
This is independent of the availability of the `stashBack` configuration.

`stashBack` should only be used from the container out.

Reply:
... aaaah, I understand I guess. We have in the same pipeline - of course - first the stash into the container and afterwards the stash outbound the container. Of course we must be able to support different filters, only as default we use the same filters. I agree, it is simpler to distinguish based on a flag.

I guess my issue "behind" is: We have some redundant code here: stashName is the same, the stash command itself it the same, the /\*commented\*/ `useDefaultExcludes` is also the same. I personally would prefer for the sake of readability something like that:

```
def includes, excludes

if(stashBack) {
  includes = config.stashIncludes.stashBack ?: config.stashIncludes.workspace
  excludes = config.stashExcludes.stashBack ?: config.stashExcludes.workspace
} else {
  includes: config.stashIncludes.workspace,
  excludes: config.stashExcludes.workspace,
}

stash(
  name: stashName,
  includes: includes
  excludes: excludes,
  //inactive due to negative side-effects, we may require a dedicated git stash to be used
  //useDefaultExcludes: false)

Reply:
Thanks, I adapted it accordingly ...